modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/DefaultExecutionContext.java

```diff
@@ -181,8 +183,17 @@ public CoreEvent getEvent() {
   @Override
   public void changeEvent(CoreEvent updated) {
     requireNonNull(event);
-    event = updated;
     securityContext = event.getSecurityContext();
```

Comment:
should get this securityContext from `updated` rather than `event`

Reply:
Nice catch :)